src/js/components/BreadcrumbComponent.jsx

```diff
@@ -12,12 +17,135 @@ var BreadcrumbComponent = React.createClass({
     taskId: React.PropTypes.string
   },
 
-  shouldComponentUpdate: function (nextProps) {
+  getInitialState: function () {
+    return {
+      collapsed: false,
+      availableWidth: null,
+      expandedWidth: null,
+      mutationObserver: null
+    };
+  },
+
+  componentDidMount: function () {
+    // Avoid referencing window from Node context
+    if (global.window != null) {
+      window.addEventListener("resize", this.handleResize);
+      window.addEventListener("focus", this.handleResize);
+      this.startMutationObserver();
+    }
+  },
+
+  componentDidUpdate: function () {
+    this.updateDimensions();
+  },
+
+  componentWillUnmount: function () {
+    if (global.window != null) {
+      window.removeEventListener("resize", this.handleResize);
+      window.removeEventListener("focus", this.handleResize);
+      this.stopMutationObserver();
+    }
+  },
+
+  shouldComponentUpdate: function (nextProps, nextState) {
+    if (nextState.availableWidth == null || nextState.expandedWidth == null) {
+      this.updateDimensions();
+      return false;
+    }
+    if (this.state.collapsed !== nextState.collapsed) {
+      return true;
+    }
+    return this.didPropsChange(nextProps);
+  },
+
+  didPropsChange: function (nextProps) {
     return nextProps.appId !== this.props.appId ||
       nextProps.groupId !== this.props.groupId ||
       nextProps.taskId !== this.props.taskId;
   },
 
+  startMutationObserver: function () {
+    if (window.MutationObserver == null) {
+      return;
+    }
+    var mutationObserver = new MutationObserver(this.handleMutation);
+    mutationObserver.observe(this.getDOMNode(), {
+      attributes: true, childList: true
+    });
+    this.setState({mutationObserver: mutationObserver});
+  },
+
+  stopMutationObserver: function () {
+    var mutationObserver = this.state.mutationObserver;
+    if (mutationObserver == null) {
+      return;
+    }
+    mutationObserver.disconnect();
+  },
+
+  handleMutation: _.throttle(function () {
```

Comment:
Works much smoother without throttling.
Why is this needed?
There isn't unnecessary rerendering even without throttling, it performs.

BTW "Trottel" is not a nice word in German.. ;)